src/rust/ide/src/controller/graph.rs

```diff
@@ -783,127 +778,102 @@ mod tests {
     }
 
     impl MockData {
-        pub fn new(code:impl Str) -> Self {
+        /// Creates a mock data with the `main` function being an inline definition with a single
+        /// node.
+        pub fn new() -> Self {
             MockData {
                 module_path  : model::module::Path::from_mock_module_name("Main"),
-                graph_id     :  Id::new_plain_name("main"),
+                graph_id     : Id::new_plain_name("main"),
                 project_name : "MockProject".to_string(),
-                code         : code.into(),
+                code         : "main = 2 + 2".to_string(),
             }
         }
 
         /// Creates a mock data with the main function being an inline definition.
         ///
         /// The single node's expression is taken as the argument.
-        pub fn new_inline(main_body:impl Str) -> Self {
-            Self::new(format!("main = {}", main_body.as_ref()))
+        pub fn new_inline(main_body:impl Into<String>) -> Self {
+            MockData {
+                code : format!("main = {}", main_body.into()),
+                ..Self::new()
+            }
         }
 
-        /// Creates module and graph controllers.
-        pub fn create_controllers(&self) -> (controller::Module,Handle) {
-            let ls = language_server::Connection::new_mock_rc(default());
-            self.create_controllers_with_ls(ls)
+        pub fn module_data(&self) -> model::module::test::MockData {
+            model::module::test::MockData {
+                code : self.code.clone(),
+                path : self.module_path.clone(),
+                ..default()
+            }
```

Comment:
I wonder... perhaps there should be rather someting like "MockData" inheriting? Or just one MockData for all structures?

Reply:
Agreed to leave this for future improvement.